alf/networks/encoding_networks.py

```diff
@@ -232,6 +232,129 @@ def __init__(self,
+    upsampling+conv might help reduce checkerboard artifacts that are common in
+    the outputs by transposed convolutions.
+    """
+
+    def __init__(self,
+                 input_size: int,
+                 upsample_conv_layer_params: Tuple[Union[int, Tuple[int]]],
+                 start_decoding_size: Union[int, Tuple[int]],
+                 start_decoding_channels: int,
+                 upsampling_mode: str = 'nearest',
+                 same_padding: bool = False,
+                 activation: Callable = torch.relu_,
+                 kernel_initializer: Callable = None,
+                 output_activation: Callable = torch.tanh,
+                 name: str = "ImageDecodingNetworkV2"):
+        """An example network of upsampling+conv for decoding images.
+
+        .. code-block:: python
+
+            net = ImageDecodingNetworkV2(input_size=100,
+                                         start_decoding_size=10,
+                                         start_decoding_channels=8,
+                                         same_padding=True,
+                                         upsample_conv_layer_params=(
+                                            2,
+                                            (16, 3, 1),
+                                            (32, 3, 1),
+                                            2,
+                                            (64, 3, 1),
+                                            (3, 3, 1)))
+            # The image shape: (8,10,10) -> (8,20,20) -> (16,20,20) -> (32,20,20)
+            #                  -> (32,40,40) -> (64,40,40) -> (3,40,40)
+
+        Args:
+            input_size: the size of the input latent vector
+            upsample_conv_layer_params: a tuple of ints or tuples. If the element
+                is an int, it represents the scaling factor for a ``torch.nn.Upsample``
+                layer; otherwise it should a tuple of ints representing conv params
+                ``(num_filters, kernel_size, strides, padding)``,
+                where ``padding`` is optional.
+            start_decoding_size: the initial height and width we'd like to have
+                for the feature map.
+            start_decoding_channels: the initial number of channels we'd
+                like to have for the feature map. Note that we always first
+                project an input latent vector into a vector of an appropriate
+                length so that it can be reshaped into (``start_decoding_channels``,
+                ``start_decoding_height``, ``start_decoding_width``).
+            upsampling_mode: the argument for choosing an upsampling algorithm
+                for ``torch.nn.Upsample``.
+            same_padding: similar to TF's conv2d ``same`` padding mode. If
+                True, the user provided paddings in ``transconv_layer_params`` will
+                be replaced by automatically calculated ones; if False, it
```

Comment:
It's better to simply describe how the padding is calculated for same_padding=True and False

Reply:
> It's better to simply describe how the padding is calculated for same_padding=True and False

``ImageEncodingNetwork`` already has a paragraph explaining the two padding modes. I've added a sentence referring to its docstring.